src/Escaper/Escaper.c

```diff
@@ -90,7 +89,15 @@ Escaper_decode_packet(Escaper* const self, uint8_t* buffer, const size_t length,
                     self->m_decoded_packet_buffer_index = 0;
                     self->m_parse_state = Escaper_State_Wait;
                 } else {
-                    self->m_decoded_packet_buffer[self->m_decoded_packet_buffer_index] = buffer[i];
+                    if(buffer[i] == ESC_START_BYTE) {
```

Comment:
This function seems to be growing... consider using switches or splitting it into smaller functions

Reply:
done